### PR TITLE
DOC-12187 File based rebalance now supports placing indexes on nodes

### PR DESCRIPTION
**Pull request by ggray-cb**
This PR addresses the doc for [MB-61771](https://issues.couchbase.com/browse/MB-61771).

Changes made in this PR, with links to preview site:

- Entry in the [What's New](https://preview.docs-test.couchbase.com/index-placement-w-file-rebalance/server/current/introduction/whats-new.html#index_762) page stating you can now select which Index Service nodes contain an index when you enable file-based index rebalance.
- Updated the Index [Rebalance Methods](https://preview.docs-test.couchbase.com/index-placement-w-file-rebalance/server/current/learn/clusters-and-availability/rebalance.html#index-rebalance-methods) section of the documentation:
  - Changed text to explain that there was a restriction on placing indexes in 7.6.0 and 7.6.1 when file-based rebalance was enabled but that is no longer the case. This explanation is flagged so it will automatically disappear in versions greater than 7.6.
  - Added some additional information about file-based rebalance. Specifically, explain the default behavior of Capella, because the Capella docs point to this page for information about rebalance. 
  - Added explanation of when the first file-based rebalance will take place for indexes created after file-based indexing is enabled.
  - Linked to sections of the doc that tell you how to enable file-based rebalance.
- Updated the description of `enableShardAffinity` in the [Set GSI Settings](https://preview.docs-test.couchbase.com/index-placement-w-file-rebalance/server/current/rest-api/post-settings-indexes.html#body-parameters) page. 

Related changes that are not in the PR (because they are in another repo):

- Updated [CREATE INDEX](https://preview.docs-test.couchbase.com/index-placement-w-file-rebalance/server/current/n1ql/n1ql-language-reference/createindex.html#index-with) reference page with statement that you can now use WITH clause to place indexes on specific nodes when file-based index rebalancing is enabled. States this was not the case in 7.6.0 and 7.6.1.
- Same update made to [ALTER INDEX](https://preview.docs-test.couchbase.com/index-placement-w-file-rebalance/server/current/n1ql/n1ql-language-reference/alterindex.html#index-with). 

There will be a separate PR that covers these changes.

### CODE REVIEW

File: modules/introduction/partials/new-features-76_2.adoc [hunk 1]
```diff
@@ -9,3 +9,10 @@
 ** xref:rest-api:backup-get-task-info.adoc[`/api/v1/cluster/self/repository/{repo-state}/{task-name}/taskHistory`]
 ** xref:rest-api:backup-get-plan-info.adoc[`/api/v1/plan/`]
 
+[#index_762]
+=== Index Service
+
+* In Couchbase Server Versions 7.6.0 and 7.6.1, enabling file-based index rebalance prevented controlling which Index Service nodes contain an index.
```
Comment (bfavini) [inline modules/introduction/partials/new-features-76_2.adoc]: Kind of hard to parse...

Maybe ....prevented you from controlling...

**Review by bfavini — APPROVED** [review]

**Comment by ggray-cb** [comment]
Casting a wider net here as this PR needs to be reviewed ASAP.

**Comment by deepkaran** [comment]
@VarunVelamuri , can you please review this one? I think the Alter Index restrictions are still applicable, right? The change has only been done for Create Index.

File: modules/learn/pages/clusters-and-availability/rebalance.adoc [hunk 2]
```diff
@@ -150,14 +151,35 @@ The rebalance process takes into account the nodes' CPU, RAM, and disk bandwidth
 By default, Couchbase Server rebuilds the indexes in their new locations during a rebalance.  
 If your cluster uses xref:learn:services-and-indexes/indexes/storage-modes.adoc#standard-index-storage[Standard Index Storage], you can choose to enable file-based rebalance which moves index files between nodes instead of rebuilding them. 
 Copying the index files is faster than having the target node rebuild the index from scratch.
-However, if you enable file-based rebalancing, you lose the ability to choose which Index Service node handles a specific index.  
+You cannot use file-based rebalance if you have enabled xref:learn:services-and-indexes/indexes/storage-modes.adoc#memory-optimized-index-storage[Memory Optimized Index Storage] because it does not store index data in files.
```
Comment (VarunVelamuri) [inline modules/learn/pages/clusters-and-availability/rebalance.adoc]: > it does not store index data in files

it does not support storing metadata in the index files which is a pre-requisite for file based rebalance

File: modules/learn/pages/clusters-and-availability/rebalance.adoc [hunk 3]
```diff
@@ -150,14 +151,35 @@ The rebalance process takes into account the nodes' CPU, RAM, and disk bandwidth
...
-After this initial rebalance, all later rebalances move the index files as long you as you keep file-base rebalance enabled.
+Couchbase Capella defaults to file-based index rebalance.
 
-If a file-base rebalance fails, you can try starting a new rebalance. 
+To learn how to enable file-based rebalance via the Couchbase Server Web Console, see File Transfer Based Rebalance under xref:manage:manage-settings/general-settings.adoc#index-storage-mode[Index Storage Mode]. To learn how to enable it via the REST API, see the `enableShardAffinity` parameter in xref:rest-api:post-settings-indexes.adoc[].
+
+// Remove after the 7.x series.
+ifeval::['{page-component-version}' == '7.6'] 
+NOTE: In Couchbase Server versions 7.6.0 and 7.6.1, when you enabled file-based rebalance you could not choose which Index Service nodes would contain an index.
+Couchbase Server 7.6.2 and later no longer have this restriction.
```
Comment (VarunVelamuri) [inline modules/learn/pages/clusters-and-availability/rebalance.adoc]: > Couchbase Server 7.6.2 and later no longer have this restriction.

The restriction is removed only for CREATE INDEX command. ALTER INDEX command still goes with the same restrictions as 7.6.0 and 7.6.1

File: modules/rest-api/pages/post-settings-indexes.adoc [hunk 4]
```diff
@@ -35,7 +35,16 @@ The parameters are specified as key-value pairs (e.g `key=value`).
...
-| Selects the xref:learn:clusters-and-availability/rebalance.adoc#index-rebalance-methods[index rebalance method]. When set to the default `false` value, Index Servers rebuild any index that are newly assigned to them during a rebalance. When set to `true`, Couchbase Server moves a reassigned index's files between Index Servers.  
+a| Selects the xref:learn:clusters-and-availability/rebalance.adoc#index-rebalance-methods[index rebalance method]:
+
+* `false` (default value): Index Service nodes rebuild indexes that are newly assigned to them during a rebalance.
+* `true`: Couchbase Server moves a reassigned index's files between Index Service nodes.  
+
+// Remove after the 7.x series.
+ifeval::['{page-component-version}' == '7.6'] 
+NOTE: In Couchbase Server versions 7.6.0 and 7.6.1, when you enabled file-based rebalance by setting `enableShardAffinity` to `true`, you could not choose which Index Service nodes would contain an index.
+Couchbase Server 7.6.2 and later no longer have this restriction.
```
Comment (VarunVelamuri) [inline modules/rest-api/pages/post-settings-indexes.adoc]: >Couchbase Server 7.6.2 and later no longer have this restriction.

Same as above. The restriction is removed only for CREATE INDEX codepaths

**Comment by VarunVelamuri** [comment]
@ggray-cb Left a few comments. Thanks!
